### PR TITLE
Feature/mailgun appender

### PR DESCRIPTION
**Pull request by chewax**
I fixed a problem where mailgun dependency could not initialize. Now it is required inside configuration function.

### CODE REVIEW

File: lib/appenders/mailgun.js [hunk 1]
```diff
@@ -17,11 +16,12 @@ function mailgunAppender(_config, _layout) {
...
             subject: _config.subject,
             text: layout(loggingEvent, config.timezoneOffset)
-        };
+        }
 
         mailgun.messages().send(data, function (error, body) {
-            if (error != null) console.error('log4js.mailgunAppender - Error happened', error);
-        });
+            if (error != null) console.error("log4js.mailgunAppender - Error happened", error);
+            else console.log("Success", body);
```
Comment (nomiddlename) [inline lib/appenders/mailgun.js]: You don't really want things going to `console.log` every time you send something via this appender, do you?

**Comment by nomiddlename** [comment]
Could you fix the conflicts and update the PR, please? You probably just need to update your fork + branch from mine.

**Comment by chewax** [comment]
Done. Please check if all is OK now.
Agree, i dont want console.log every time you send something via this appender. I removed it.

**Comment by nomiddlename** [comment]
Published to npm in version 0.6.36

**Pull request merged**